src/bootstrap/flags.rs

```diff
@@ -468,7 +472,7 @@ Arguments:
                 .into_iter()
                 .map(|p| p.into())
                 .collect::<Vec<_>>(),
-            warnings: matches.opt_str("warnings").map(|v| v == "deny"),
+            deny_warnings,
```

Comment:
IMO readability is slightly improved by inlining the `let`:
`deny_warnings: parse_deny_warnings(&matches)`
Not a blocker though.